src/shared/components/query/GeneSetSelector.tsx

```diff
@@ -60,18 +64,14 @@ export default class GeneSetSelector extends QueryStoreComponent<{}, {}>
 	    ];
 	}
 
-	@computed get textAreaRef()
-	{
-		if (this.store.geneQueryErrorDisplayStatus === 'shouldFocus')
-			return (textArea:HTMLTextAreaElement) => {
-				let {error} = this.store.oql;
-				if (textArea && error)
-				{
-					textArea.focus();
-					textArea.setSelectionRange(error.start, error.end);
-					this.store.geneQueryErrorDisplayStatus = 'focused';
-				}
-			};
+	@autobind
+	geneInputUpdate(
+		oql: {query: SingleGeneQuery[], error?: {start: number, end: number, message: string}},
+		genes: {found: Gene[], suggestions: GeneReplacement[]},
+		queryStr: string,
+	): void {
+		this.store.geneQuery = queryStr;
```

Comment:
Whenever we mutate mobx observables, we need to do so inside an "action."   Doing so allows for batch updates.  Setting multiple values (as you are doing here) will only trigger one update "signal" to observers when all changes are made.  Even when we're only making a single change, we should do it inside an action because it can help for debugging. 
You can do this here by adding @action decorator on this method.

Reply:
FYI, this rule was not obeyed in the previous code here